addons/web_editor/models/ir_ui_view.py

```diff
@@ -80,10 +85,71 @@ def save_oe_structure(self, el):
             'mode': 'extension',
         }
         vals.update(self._save_oe_structure_hook())
-        self.env['ir.ui.view'].create(vals)
+        oe_structure_view = self.env['ir.ui.view'].create(vals)
+        self._copy_custom_snippet_translations(oe_structure_view, 'arch_db')
 
         return True
 
+    @api.model
+    def _copy_custom_snippet_translations(self, record, html_field):
+        """ Given a ``record`` and its HTML ``field``, detect any
+        usage of a custom snippet and copy its translations.
+        """
+        lang_value = record[html_field]
+        if not lang_value:
+            return
+
+        tree = html.fromstring(lang_value)
+        for custom_snippet_el in tree.xpath('//*[hasclass("s_custom_snippet")]'):
+            custom_snippet_name = custom_snippet_el.get('data-name')
+            custom_snippet_view = self.search([('name', '=', custom_snippet_name)], limit=1)
+            if custom_snippet_view:
+                self._copy_model_field_terms_translations(custom_snippet_view, 'arch_db', record, html_field)
+
+    @api.model
+    def _copy_model_field_terms_translations(self, record_from, name_field_from, record_to, name_field_to):
+        """ Copy the terms translation from a record/field ``Model1.Field1``
+        to a (possibly) completely different record/field ``Model2.Field2``.
+
+        For instance, copy the translations of a
+        ``product.template.html_description`` field to a ``ir.ui.view.arch_db``
+        field.
+
+        The method takes care of read and write access of both records/fields.
+        """
+        record_to.check_access_rights('write')
+        record_to.check_access_rule('write')
+
+        # This will also implicitly check for `read` access rights
+        if not record_from[name_field_from] or not record_to[name_field_to]:
+            return
+
+        field_from = record_from._fields[name_field_from]
+        field_to = record_to._fields[name_field_to]
+        error_callable_msg = "'translate' property of field %r is not callable"
+        if not callable(field_from.translate):
+            raise ValueError(error_callable_msg % field_from)
+        if not callable(field_to.translate):
+            raise ValueError(error_callable_msg % field_to)
+
+        lang_env = self.env.lang or 'en_US'
+        langs = set(lang for lang, _ in self.env['res.lang'].get_installed())
+
+        # 1. Get translations
+        translation_dictionary = field_from.get_translation_dictionary(
+            record_from[name_field_from],
+            {lang: record_from.with_context(prefetch_langs=True, lang=lang)[name_field_from] for lang in langs if lang != lang_env}
+        )
+
+        # 2. Set translations
+        new_value = {
+            lang: field_to.translate(lambda term: translation_dictionary.get(term, {}).get(lang), record_to[name_field_to])
+            for lang in langs
+        }
```

Comment:
@HydrionBurst Maybe here I could force `en_US` if not installed on the DB? With `record_to[name_field_to]` as value.

Doesn't seems to be that problematic to not have an en_US value tho